cluster-autoscaler/cloudprovider/POLICY.md

```diff
@@ -0,0 +1,127 @@
+    so that they can approve PRs to their cloudprovider.
+  * Cloudprovider shouldn't introduce new dependencies (such as clients/SDKs)
+    to top-level go.mod vendor, unless those dependencies are already imported
+    by kubernetes/kubernetes repository and the same version of the library is
+    used by CA and Kubernetes. This requirement is mainly driven by
+    the problems with version conflicts in transitive dependencies we've
+    experienced in the past.
+
+Note: Any functions in cloud\_provider.go marked as 'Implementation optional'
+may be left unimplemented. Those functions provide additional functionality, but
+are not critical. To leave a function unimplemented just have it return
+cloudprovider.ErrNotImplemented.
+
+### External provider
+
+An alternative to implementing an in-tree cloudprovider is to use existing
+[External
+gRPC](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/externalgrpc)
+provider. Integrating with gRPC interface may be easier than implementing an
+in-tree cloudprovider and the gRPC provider comes with some essential caching
+built in.
+
+An external cloudprovider implementation doesn't live in this repository and is
+not a part of CA image. As such it is also not a subject to this policy.
+
+## Cloudprovider maintenance requirements
+
+In order to allow code changes to Cluster Autoscaler that would require
+non-trivial changes in cloudproviders this policy introduces _Cloudprovider
+maintenance request_ (CMR) mechanism.
+
+ * CMR will be issued via a github issue tagging all
+   cloudprovider owners and describing the problem being solved and the changes
+   requested.
+ * CMR will clearly state the minor version in which the changes are expected
+   (ex. 1.26).
+ * CMR will need to be discussed on sig-autoscaling meeting and approved by
+   sig leads before being issued. It will also be announced on sig-autoscaling
+   slack channel and highlited in sig-autoscaling meeting notes.
+ * A CMR may be issued no later then [enhancements
+   freeze](https://github.com/kubernetes/sig-release/blob/master/releases/release_phases.md#enhancements-freeze)
+   of a given Kubernetes minor version.
+
+Cloudprovider owners will be required to address CMR or request an exception via
+the CMR github issue. A failure to take any action will result in cloudprovider
+being considered abandoned and marking it as deprecated as described below.
+
+### Empty maintenance request
+
+If no CMRs are issued in a given minor release, core maintainers will issue an
+_empty CMR_. The purpose of an empty CMR is to verify that cloudprovider owners
+are still actively maintaining their integration. The only action required for
+an empty CMR is replying on the github issue. Only one owner from each
+cloudprovider needs to reply on the issue.
+
+Empty CMR follows the same rules as any other CMR. In particular it needs to be
+issued by enhancements freeze.
```

Comment:
i think it would be nice to bring this part on a separate review since it proposes a new process.